File: csharp/DesignPattern.Command/Waiter.cs

using System;
using System.Collections.Generic;
namespace DesignPattern.Command
{
    /// <summary>
    /// 服务员
    /// </summary>
    public class Waiter
    {
        private List<Command> commands;

        public Waiter()
        {
            this.commands = new List<Command>();
        }

        public void SetOrder(Command command)
        {
            this.commands.Add(command);
            Console.WriteLine("增加订单：" + command.Name);
        }

        public void CancelOrder(Command command)
        {
            this.commands.Remove(command);
            Console.WriteLine("取消订单：" + command.Name);
        }

        public void Notify()
        {
            foreach (Command command in this.commands)
            {
                command.Cook();
            }
        }
    }
}

File: java/src/com/zhy/DesignPattern/Command/Waiter.java
package com.zhy.DesignPattern.Command;

import java.util.ArrayList;
import java.util.List;

public class Waiter {

    private List<AbstractCommand> commands;

    public Waiter()
    {
        this.commands = new ArrayList<AbstractCommand>();
    }

    public void SetOrder(AbstractCommand command)
    {
        this.commands.add(command);
        System.out.println("增加订单：" + command.getName());
    }

    public void CancelOrder(AbstractCommand command)
    {
        this.commands.remove(command);
        System.out.println("取消订单：" + command.getName());
    }

    public void Notify()
    {
        for (AbstractCommand command : this.commands)
        {
            command.Cook();
        }
    }
}
